src/common/Graph.js

```diff
@@ -1,77 +1,122 @@
 import { PriorityQueue } from './PriorityQueue.js'
+import Victor from 'victor'
 
-export const mix = (u, v, s) => {
+export const mix = (v1, v2, s) => {
+  let u = [v1.x, v1.y]
+  let v = [v2.x, v2.y]
   var result = []
+
   for (var i = 0; i < u.length; ++i ) {
     result.push((1.0 - s) * u[i] + s * v[i])
   }
 
-  return result
-}
-
-export const vec2 = (...args) => {
-  switch (args.length) {
-    case 0: args.push(0.0); break
-    default: args.push(0.0)
-  }
-
-  return args.splice(0, 2)
+  return new Victor(result[0], result[1])
 }
 
+// note: requires string-based nodes to work properly
 export default class Graph {
```

Comment:
I don't want to make things too complicated, and make work for the sake of work. But would it be interesting to try to make some of this Graph stuff into it's own npm library? I think it would add some complication, especially when trying to update it, but it might be something to think about.

Reply:
I have been looking at changing over to this library: https://github.com/trekhleb/javascript-algorithms. I keep having to find and add various graph algorithms, and he's written them all. As you said, it's more work though... I have my sights set on a new shape which needs may push me in this direction :)